packages/tools/devtools/devtools-core/src/data-visualization/DefaultVisualizers.ts

```diff
@@ -245,13 +246,28 @@ export const visualizeSharedTree: VisualizeSharedObject = async (
 	visualizeChildData: VisualizeChildData,
 ): Promise<FluidObjectTreeNode> => {
 	const sharedTree = sharedObject as ISharedTree;
-	const content = sharedTree.contentSnapshot();
+	const contentSnapshot = sharedTree.contentSnapshot();
+
+	// Root node of the SharedTree's treeview. Assume there is only one root node.
+	const treeView = contentSnapshot.tree[0];
+
+	// Schema of the tree node.
+	const treeSchema = contentSnapshot.schema.nodeSchema.get(treeView.type);
+
+	// Traverses the SharedTree and generates a visual representation of the tree (mainly composed of `schema` and `fields` field).
+	const visualTreeRepresentation = visualizeSharedTreeNodeBySchema(
+		treeView,
+		treeSchema,
+		contentSnapshot,
+	);
+
+	// Maps the `visualTreeRepresentation` in the format compatible to {@link visualizeChildData} function.
+	const visualTree = toVisualTree(visualTreeRepresentation);
 
 	return {
 		fluidObjectId: sharedTree.id,
 		children: {
-			tree: await visualizeChildData(content.tree),
-			schema: await visualizeChildData(encodeTreeSchema(content.schema)),
+			tree: visualTree,
```

Comment:
Nit: This extra layer in the visual tree is probably not needed. We could flatten it a bit by doing something like:
```typescript
return {
	...visualTree,
	fluidObjectId: sharedTree.id,
	typeMetadata: "SharedTree",
	nodeKind: VisualNodeKind.FluidTreeNode,
}
```

Might be worth experimenting with, at least.

Reply:
I think the `children` field is still needed. The `visualizeSharedTree()` functions returns the promise of `FluidObjectTreeNode` which extends the `TreeNodeBase` which then requires the `children` field.

Reply:
```
export const visualizeSharedTree: VisualizeSharedObject = async (
	sharedObject: ISharedObject,
	visualizeChildData: VisualizeChildData,
): Promise<VisualChildNode> => {
	const sharedTree = sharedObject as ISharedTree;
	const contentSnapshot = sharedTree.contentSnapshot();

	// Root node of the SharedTree's treeview. Assume there is only one root node.
	const treeView = contentSnapshot.tree[0];

	// Schema of the tree node.
	const treeSchema = contentSnapshot.schema.nodeSchema.get(treeView.type);

	// Traverses the SharedTree and generates a visual representation of the tree and its schema.
	const visualTreeRepresentation = visualizeSharedTreeNodeBySchema(
		treeView,
		treeSchema,
		contentSnapshot,
	);

	// Maps the `visualTreeRepresentation` in the format compatible to {@link visualizeChildData} function.
	const visualTree = toVisualTree(visualTreeRepresentation);

	return {
		...visualTree,
		fluidObjectId: sharedTree.id,
		typeMetadata: "SharedTree",
		nodeKind: VisualNodeKind.TreeNode,
	};
};
```

If we change this to type Promise<VisualChildNode>, we wouldn't need a `fluidObjectId`?